src/task.py

```diff
@@ -0,0 +1,195 @@
+    """
+    old_env = os.environ.copy()
+    original_aws_key_id = os.environ.pop("AWS_ACCESS_KEY_ID", None)
+    original_aws_key_secret = os.environ.pop("AWS_SECRET_ACCESS_KEY", None)
+    original_session = os.environ.pop("AWS_SESSION_TOKEN", None)
+    try:
+        gdal_opts = {**GDAL_CLOUD_DEFAULTS, **kwargs}
+        gdal_opts.update(
+            {
+                "GDAL_HTTP_TCP_KEEPALIVE": "YES",
+                "AWS_S3_ENDPOINT": "eodata.dataspace.copernicus.eu",
+                "AWS_HTTPS": "YES",
+                "AWS_VIRTUAL_HOSTING": "FALSE",
+                "GDAL_HTTP_UNSAFESSL": "YES",
+            }
+        )
+
+        use_profile = (
+            os.path.exists(os.path.expanduser("~/.aws/config")) and not force_keys
+        )
+
+        if use_profile:
+            logger.debug("Using copernicus profile")
+            session = boto3.Session(profile_name=profile_name)
+        else:
+            logger.debug("Not using copernicus profile pulling keys")
+            copernicus_access_key = (
+                os.environ.get("CDSE_AWS_ACCESS_KEY_ID") or original_aws_key_id
+            )
+            copernicus_secret_key = (
+                os.environ.get("CDSE_AWS_SECRET_ACCESS_KEY") or original_aws_key_secret
+            )
+
+            if not copernicus_access_key or not copernicus_secret_key:
+                raise RuntimeError(
+                    "Copernicus credentials not found. "
+                    "Set CDSE_AWS_ACCESS_KEY_ID and CDSE_AWS_SECRET_ACCESS_KEY "
+                    "or configure ~/.aws/config profile 'copernicus'."
+                )
+
+            session = boto3.Session(
+                aws_access_key_id=copernicus_access_key,
+                aws_secret_access_key=copernicus_secret_key,
+                aws_session_token=original_session,
+            )
+
+        aws = AWSSession(session)
+        with rio.Env(aws_session=aws, **gdal_opts) as env:
+            yield env
+
+    finally:
+        os.environ.clear()
+        os.environ.update(old_env)
+
+
+class CopernicusReadAwsStacTask(AwsStacTask):
+    """
+    Reads inputs using Copernicus creds+endpoint, writes outputs using default/original creds.
+    """
+
+    def __init__(
+        self,
+        *args,
+        copernicus_profile="copernicus",
+        force_keys=False,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.copernicus_profile = copernicus_profile
+        self.logger: Logger = kwargs.get("logger", getLogger())
+
+    def run(self):
+        # ---- READ PHASE (temp copernicus session) ----
+        aws, boto_session, gdal_opts = get_copernicus_rio_config()
+        # using raserio env probably unnecessary as configure_rio is actually passing down the config to the dask client
+
+        configure_rio(cloud_defaults=True, aws={"session": boto_session}, **gdal_opts)
+        items = self.searcher.search(self.area)
+        input_data = self.loader.load(items, self.area)
```

Comment:
You possibly just want to do `.compute()` here and load all the data into memory. It's not much data, so it's fine.